plugins/outputs/influxdb_v2/batching.go

```diff
@@ -102,11 +103,23 @@ func (b *batch) split() (first, second *batch) {
 }
 
 func (b *batch) serialize(serializer ratelimiter.Serializer, limit int64, encoder internal.ContentEncoder) (int64, error) {
-	// Serialize the metrics with the remaining limit, exit early if nothing was serialized
+	// Serialize the metrics with the remaining limit,
 	body, serr := serializer.SerializeBatch(b.metrics, limit)
 	if serr != nil && !errors.Is(serr, internal.ErrSizeLimitReached) {
-		return int64(len(body)), serr
+		// When only part of the metrics failed to be serialized we should remove
+		// them from the normal handling and mark them as rejected for upstream
+		// to pass on this information
+		var werr *internal.PartialWriteError
+		if errors.As(serr, &werr) {
+			for i, idx := range slices.Backward(werr.MetricsReject) {
+				werr.MetricsReject[i] = b.indices[idx]
```

Comment:
Is the reversing of the order intentional? I'd imagine that the rejected metrics would still be in order; but slices.Backward reverses it?

(just a comment explaining logic is fine here)

Reply:
Yeah, I was tripping on the index logic quite a few times, maybe it's just me though 😅

Reply:
Ok I double checked the code and we do not enforce an order on the indices as the buffers do things robustly already. So no additional change required...